deanonymization/check_bitcoin_addresses.go

```diff
@@ -86,9 +86,54 @@ func ValidA58(a58 []byte) (ok bool) {
 	return a.EmbeddedChecksum() == a.ComputeChecksum()
 }
 
+// P37 is a type for a 37 byte (not base58 encoded) private key.
+type P37 [37]byte
+
+// Version extracts the version byte from a private key
+func (a *P37) Version() byte {
+	return a[0]
+}
+
+// Set58 takes a base58 encoded private key and decodes it into the receiver.
+// Errors are returned if the argument is not valid base58 or if the decoded
+// value does not fit in the 37 byte private key.  The private key is not otherwise
+// checked for validity.
+func (a *P37) Set58(s []byte) error {
```

Comment:
it's good practice to call the label the variable after the type in this case `a *P37` should be `p * P37`